Tasks/AzureFunctionAppContainerV1/azurefunctiononcontainerprovider.ts

```diff
@@ -37,38 +40,55 @@ export class AzureFunctionOnContainerDeploymentProvider{
 
         this.appService = new AzureAppService(this.azureEndpoint, this.taskParams.ResourceGroupName, this.taskParams.WebAppName, this.taskParams.SlotName);
         this.appServiceUtility = new AzureAppServiceUtility(this.appService);
+        this.appServiceUtilityExt = new AzureAppServiceUtilityExt(this.appService);
         this.taskParams.isLinuxContainerApp = true;
-        this.kuduService = await this.appServiceUtility.getKuduService();
-        this.kuduServiceUtility = new KuduServiceUtility(this.kuduService);
+        this.isCentauri = await this.appServiceUtilityExt.isFunctionAppOnCentauri();
+        if (!this.isCentauri){
+            this.kuduService = await this.appServiceUtility.getKuduService();
+            this.kuduServiceUtility = new KuduServiceUtility(this.kuduService);
+        }
         tl.debug(`Resource Group: ${this.taskParams.ResourceGroupName}`);
         tl.debug(`is Linux container web app: ${this.taskParams.isLinuxContainerApp}`);
     }
 
     public async DeployWebAppStep() {
         tl.debug("Performing container based deployment.");
-
         let containerDeploymentUtility: ContainerBasedDeploymentUtility = new ContainerBasedDeploymentUtility(this.appService);
-        await containerDeploymentUtility.deployWebAppImage(this.taskParams);
-        let linuxFunctionStorageSetting: string = ''; 
-        if (!this.taskParams.AppSettings || this.taskParams.AppSettings.indexOf(linuxFunctionStorageSettingName) < 0) { 
-            linuxFunctionStorageSetting = `${linuxFunctionStorageSettingName} ${linuxFunctionStorageSettingValue}`; 
-        }
-        this.taskParams.AppSettings = this.taskParams.AppSettings ? this.taskParams.AppSettings.trim() + " " + linuxFunctionStorageSetting : linuxFunctionStorageSetting;
-        let customApplicationSettings = ParameterParser.parse(this.taskParams.AppSettings);
-        await this.appServiceUtility.updateAndMonitorAppSettings(customApplicationSettings);
         
-        await this.appServiceUtility.updateScmTypeAndConfigurationDetails();
+        if (this.isCentauri){
```

Comment:
There is a lot of code duplication in following lines and it is really hard to determine what are the differences between those two if branches. Can you please try to simplify it? For example:

```ts
await containerDeploymentUtility.deployWebAppImage(this.taskParams, !this.isCentauri);
let linuxFunctionStorageSetting: string = ''; 
if (!this.taskParams.AppSettings || this.taskParams.AppSettings.indexOf(linuxFunctionStorageSettingName) < 0) { 
    linuxFunctionStorageSetting = `${linuxFunctionStorageSettingName} ${linuxFunctionStorageSettingValue}`; 
}
this.taskParams.AppSettings = this.taskParams.AppSettings ? this.taskParams.AppSettings.trim() + " " + linuxFunctionStorageSetting : linuxFunctionStorageSetting;
let customApplicationSettings = ParameterParser.parse(this.taskParams.AppSettings);

if (this.isCentauri){
 await this.appServiceUtilityExt.updateAndMonitorAppSettingsCentauri(customApplicationSettings, fal
} else {
 await this.appServiceUtility.updateAndMonitorAppSettings(customApplicationSettings);
 await this.appServiceUtility.updateScmTypeAndConfigurationDetails();            
}
```